File: apps/mobile/ios/LocationTracker/LocationAccuracyModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(LocationAccuracyModule, NSObject)

RCT_EXTERN_METHOD(isPreciseLocationEnabled:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

@end

File: apps/mobile/ios/LocationTracker/LocationAccuracyModule.swift
import CoreLocation
import React

@objc(LocationAccuracyModule)
class LocationAccuracyModule: NSObject {
  @objc
  static func requiresMainQueueSetup() -> Bool {
    true
  }

  @objc
  func isPreciseLocationEnabled(
    _ resolve: RCTPromiseResolveBlock,
    rejecter reject: RCTPromiseRejectBlock
  ) {
    let locationManager = CLLocationManager()
    resolve(locationManager.accuracyAuthorization == .fullAccuracy)
  }
}
